version/version.go

```diff
@@ -2,46 +2,84 @@ package version
 
 import (
 	"fmt"
+	"strconv"
+	"strings"
 )
 
-const (
-	VersionMajor = 2  // Major version component of the current release
-	VersionMinor = 0  // Minor version component of the current release
-	VersionPatch = 1  // Patch version component of the current release
-	VersionMeta  = "" // Version metadata to append to the version string
+var (
+	GitTag       = "" // Git tag of this release
+	versionMajor = 0  // Major version component of the current release
+	versionMinor = 0  // Minor version component of the current release
+	versionPatch = 0  // Patch version component of the current release
+	versionMeta  = "" // Meta information of the current release
 )
 
 // Version holds the textual version string.
 var Version = func() string {
```

Comment:
`Version` could be a constant by implementing this as follows:
```
// Version holds the textual version string.
const Version = GitTag

func init() {
	// in case of no tag or small/irregular tag, return it as is
	if len(GitTag) >= 2 {
		versionMajor, versionMinor, versionPatch, versionMeta = parseVersion(GitTag)
	}
}
```

In this case, `GitTag` could be removed and the make file could just set the `Version` directly. Wdyt?

Reply:
I like the idea of removing the 4 global variables, and just having them as return from `parseVersion`. So I will keep only `Version` as a package variable, which gets assigned at linktime. `GitTag` is replaced by `Version`.